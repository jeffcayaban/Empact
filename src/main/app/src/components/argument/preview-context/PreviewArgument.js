import React from 'react';
import {Button, Modal} from "react-bootstrap";
import {Route} from "react-router-dom";

/**
 * Provides a quick preview of an argument in the form of a modal.
 */

class PreviewArgument extends React.PureComponent {

    render() {
        const { parentArgument, handleHide, show, labels, renderContents } = this.props;

        return (
            <Modal show={show} onHide={handleHide}>
                <Modal.Header closeButton>
                    <Modal.Title>Argument Preview</Modal.Title>
                </Modal.Header>
                <Modal.Body>
                    { renderContents(parentArgument, labels, false) }
                </Modal.Body>
                <Modal.Footer>
                    <Route render={({ history }) => (
                        <Button bsStyle="primary" onClick={() => { history.push(`/argument/${parentArgument.id}`) }}>
                            View
                        </Button>
                    )}/>
                    <Button onClick={handleHide}>Close</Button>
                </Modal.Footer>
            </Modal>
        );
    }
}

export default PreviewArgument;
