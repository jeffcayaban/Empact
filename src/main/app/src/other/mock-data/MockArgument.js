import {CRITICAL_QUESTION_IDS, EXPERT_OPINION, ROLE_ADMIN} from "../../constants";

export default {
    id: 'testId',
    isSupporting: true,
    rootPetitionId: 'testRootPetitionId',
    parentArgumentId: 'testParentArgumentId',
    subCriticalQuestionId: null,
    agreesWithCQ: true,
    criticalQuestionId: CRITICAL_QUESTION_IDS[0],
    creationDateTime: 1551716401541,
    lastUpdatedDateTime: 1551716402000,
    createdBy: {
        id: 'testId',
        username: 'testUsername',
        firstName: 'testFirstName',
        lastName: 'testLastName',
        authorities: [ROLE_ADMIN]
    },
    argumentType: EXPERT_OPINION,
    isAnonymous: false,
    sources: [],
    expert: "testExpert",
    expertDomain: 'testDomain',
    expertAssertion: 'testAssertion'
};