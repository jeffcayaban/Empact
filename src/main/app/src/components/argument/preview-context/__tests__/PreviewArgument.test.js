import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import PreviewArgument from "../PreviewArgument";

import MockArgument from '../../../../other/mock-data/MockArgument.js';

describe('PreviewArgument', () => {

    it('should render successfully', () => {
        const component = shallow(
            <PreviewArgument
                argument={MockArgument}
                handleHide={jest.fn()}
                show={true}
                labels={['testLabel1', 'testLabel2', 'testLabel3']}
                renderContents={jest.fn()}
            />
        );

        expect(toJson(component)).toMatchSnapshot();
    });
});
