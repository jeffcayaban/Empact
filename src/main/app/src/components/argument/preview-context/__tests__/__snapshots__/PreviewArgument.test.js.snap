// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewArgument should render successfully 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[MockFunction]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      Argument Preview
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  />
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <Route
      render={[Function]}
    />
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[MockFunction]}
    >
      Close
    </Button>
  </ModalFooter>
</Modal>
`;
